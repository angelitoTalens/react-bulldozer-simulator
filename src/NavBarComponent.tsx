import React from "react";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";

type NavBarComponentProps = {
    onRestart: (event: React.MouseEvent<HTMLAnchorElement>) => void,
}

function NavBarComponent(props: NavBarComponentProps) {
    return (
        <Navbar bg="dark" variant="dark">
            <Navbar.Brand>Site Clearing Simulator</Navbar.Brand>
                <Nav>
                    <Nav.Link onClick={props.onRestart}>Restart</Nav.Link>
                </Nav>
        </Navbar>
    );
}

export default NavBarComponent