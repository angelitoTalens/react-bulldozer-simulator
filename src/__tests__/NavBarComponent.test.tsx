import React from 'react';
import renderer from 'react-test-renderer';

import NavBarComponent from '../NavBarComponent';

test ("Test NavBarComponent", ()  => {

    const component = renderer.create(
        <NavBarComponent
            onRestart={()=>{}}
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});