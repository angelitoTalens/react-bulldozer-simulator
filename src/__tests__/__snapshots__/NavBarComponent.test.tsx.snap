// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NavBarComponent 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <span
    className="navbar-brand"
  >
    Site Clearing Simulator
  </span>
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  >
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      Restart
    </a>
  </div>
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  >
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      Load File
    </a>
  </div>
</nav>
`;
